src/ott/solvers/linear/univariate.py

```diff
@@ -44,29 +44,43 @@ class UnivariateSolver:
       - `'quantile'` - Take equally spaced quantiles of the distances.
       - `'equal'` - No subsampling is performed, requires distributions to have
         the same number of points.
+      - `'wasserstein'` - Compute the `p`-norm distance between the CDFs of the
+        weighted distributions.
 
     n_subsamples: The number of samples to draw for the "quantile" or
       "subsample" methods.
+    p: The exponent for the `'wasserstein'` method.
   """
 
   def __init__(
       self,
       sort_fn: Optional[Callable[[jnp.ndarray], jnp.ndarray]] = None,
       cost_fn: Optional[costs.CostFn] = None,
-      method: Literal["subsample", "quantile", "equal"] = "subsample",
+      method: Literal["subsample", "quantile", "wasserstein",
+                      "equal"] = "subsample",
       n_subsamples: int = 100,
+      p: Optional[float] = None,
   ):
     self.sort_fn = jnp.sort if sort_fn is None else sort_fn
     self.cost_fn = costs.PNormP(2) if cost_fn is None else cost_fn
     self.method = method
     self.n_subsamples = n_subsamples
+    self.p = 2.0 if p is None else p
 
-  def __call__(self, x: jnp.ndarray, y: jnp.ndarray) -> float:
+  def __call__(
+      self,
+      x: jnp.ndarray,
+      y: jnp.ndarray,
+      a: Optional[jnp.ndarray] = None,
+      b: Optional[jnp.ndarray] = None
+  ) -> float:
     """Computes the Univariate OT Distance between `x` and `y`.
 
     Args:
       x: The first distribution of shape ``[n,]`` or ``[n, 1]``.
       y: The second distribution of shape ``[m,]`` or ``[m, 1]``.
+      a: The probability weights for `x` (only for wasserstein)
```

Comment:
Would rephrase as;
```rst
a: The first marginals when ``method = 'wasserstein'``. If :obj:`None`, uniform will be used.
```
Also in the future, this should be refactored (most likely accepting again a `LinearProblem` with `PointCloud` geometry).

Reply:
Do you mean the whole solver or this particular method?